lib/dom_parser.rb

```diff
@@ -0,0 +1,143 @@
+require_relative 'tree.rb'
+require_relative 'parse_tag.rb'
+
+class DomParser
+  
+  attr_accessor :parse_tree
+
+  def initialize
+    @parse_tree = TagTree.new
+  end
+
+  def dom_reader(html)
+    
+    if File.file?(html)
+       lines = File.readlines(html)
+       html = lines.join.gsub("\n", " ")
+    end
+    build_dom_tree(html)
+
+  end
+    
+  #Start method tokenize to create text and tag tokens
+  def tokenize(html_string)
+    
+    regex = /<.*?>/
+    tag_list = html_string.scan(regex)
+    text_list = html_string.split(regex)
```

Comment:
It looks like this splits the page by the tags, returning just the text? Very nice! I tried to do it with all sorts of crazy regexes.